tests/php/unit/blocks/migration/test-class-post-type.php

```diff
@@ -147,6 +148,97 @@ public function test_migrate_all() {
 		}
 	}
 
+	/**
+	 * Test migrate_all with many posts.
+	 *
+	 * @covers \Block_Lab\Blocks\Migration\Post_Type::migrate_all()
+	 */
+	public function test_migrate_all_many_posts() {
+		$number_of_block_lab_posts = 203;
+		$block_lab_posts           = [];
+		for ( $i = 0; $i < $number_of_block_lab_posts; $i++ ) {
+			$block_lab_posts[] = $this->create_block_post( $this->post_block_initial_content );
+		}
+
+		$non_block_lab_posts           = [];
+		$non_block_lab_post_content    = 'This is example content';
+		$number_of_non_block_lab_posts = 21;
+		for ( $i = 0; $i < $number_of_non_block_lab_posts; $i++ ) {
+			$non_block_lab_posts[] = $this->factory()->post->create_and_get(
+				[ 'post_content' => $non_block_lab_post_content ]
+			);
+		}
+
+		$this->instance->migrate_all();
+
+		$queried_block_lab_post_type = new WP_Query(
+			[
+				'post_type'      => self::PREVIOUS_POST_TYPE_SLUG,
+				'posts_per_page' => -1,
+			]
+		);
+
+		// All block_lab CPT posts should have been migrated, so there should be none here.
+		$this->assertEquals( 0, $queried_block_lab_post_type->post_count );
+
+		$queried_new_cpt_posts = new WP_Query(
+			[
+				'post_type'      => self::NEW_POST_TYPE_SLUG,
+				'posts_per_page' => -1,
+			]
+		);
+
+		// The block_lab CPT posts should now have the new post_type and migrated post_content.
+		$this->assertEquals( $number_of_block_lab_posts, $queried_new_cpt_posts->post_count );
```

Comment:
How does this test ensure that the same set of posts weren't just looped through during migration?

e.g.
initial state: `a`,`b`,`c`,`d`,`e`,`f`,`g`,`h`,`i`,`j`,`k`,`l`,`m`,`n`,`o`,`p`
end state: `a`,`b`,`c`,`a`,`b`,`c`,`a`,`b`,`c`,`a`,`b`,`c`,`a`,`b`,`c`,`a`

(both sets have a count of 16)

Reply:
Feel free to ignore this if you think it's safe enough as is.

Reply:
Hm, this tests that the right amount of posts had their `post_type` changed, as they'll have the `self::NEW_POST_TYPE_SLUG`:

https://github.com/getblocklab/block-lab/blob/f63f28b1f930122421bafd4e814505389623e969/tests/php/unit/blocks/migration/test-class-post-type.php#L184-L189

Still, that `assertEquals()`  doesn't ensure that the `post_content` was changed.